Error when compiling ocaml 4.02.1 with option "-j 0"

I got the following error when compiling ocaml 4.02.1 with the option `-j 0`. It worked fine without `-j 0` afterwards.

``` shell
$ opam switch -j 0 install 4.02.1
[compiler.get] Downloading http://caml.inria.fr/pub/distrib/ocaml-4.02/ocaml-4.02.1.tar.gz
Now compiling OCaml. This may take a while, please bear with us...
Done.
# To setup the new switch in the current shell, you need to run:
eval `opam config env`
The following actions will be performed:
 - install   base-unix.base
 - install   base-bigarray.base
 - install   base-threads.base
=== 3 to install ===

=-=- Synchronizing package archives -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

=-=- Installing packages =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] # opam-version    1.2.0 (ed04dabfc81515ec7d782f6b724df72d961e3300)
# os              linux
opam: "wait" failed: No child processes
Fatal error:
File "client/opamSolution.ml", line 165, characters 11-17: Assertion failed
```

--- a/src/client/opamArg.ml
+++ b/src/client/opamArg.ml
@@ -503,10 +503,19 @@ let repo_kind_flag =
     Arg.(some (enum (main_kinds @ aliases_kinds))) None
 
 let jobs_flag =
+  let positive : int Arg.converter =
+    let (parser, printer) = Arg.int in
+    let parser s =
+      match parser s with
+      | `Error _ as e -> e
+      | `Ok n as r -> if n <= 0
+        then `Error "expected a positive integer"
+        else r in
+    (parser, printer) in
   mk_opt ["j";"jobs"] "JOBS"
     "Set the maximal number of concurrent jobs to use. You can also set it using \
      the $(b,\\$OPAMJOBS) environment variable."
-    Arg.(some int) None
+    Arg.(some positive) None
 
 let pattern_list =
   arg_list "PATTERNS" "List of package patterns." Arg.string

--- a/src/client/opamArg.ml
+++ b/src/client/opamArg.ml
@@ -507,7 +507,7 @@ let jobs_flag =
     let (parser, printer) = Arg.int in
     let parser s =
       match parser s with
-      | `Error _ as e -> e
+      | `Error _ -> `Error "expected a positive integer"
       | `Ok n as r -> if n <= 0
         then `Error "expected a positive integer"
         else r in

--- a/src/client/opamArg.ml
+++ b/src/client/opamArg.ml
@@ -301,6 +301,16 @@ let package_name =
   let print ppf pkg = pr_str ppf (OpamPackage.Name.to_string pkg) in
   parse, print
 
+let positive_integer : int Arg.converter =
+  let (parser, printer) = Arg.int in
+  let parser s =
+    match parser s with
+    | `Error _ -> `Error "expected a positive integer"
+    | `Ok n as r -> if n <= 0
+      then `Error "expected a positive integer"
+      else r in
+  (parser, printer)
+
 (* name * version option *)
 let package =
   let parse str =
@@ -503,19 +513,10 @@ let repo_kind_flag =
     Arg.(some (enum (main_kinds @ aliases_kinds))) None
 
 let jobs_flag =
-  let positive : int Arg.converter =
-    let (parser, printer) = Arg.int in
-    let parser s =
-      match parser s with
-      | `Error _ -> `Error "expected a positive integer"
-      | `Ok n as r -> if n <= 0
-        then `Error "expected a positive integer"
-        else r in
-    (parser, printer) in
   mk_opt ["j";"jobs"] "JOBS"
     "Set the maximal number of concurrent jobs to use. You can also set it using \
      the $(b,\\$OPAMJOBS) environment variable."
-    Arg.(some positive) None
+    Arg.(some positive_integer) None
 
 let pattern_list =
   arg_list "PATTERNS" "List of package patterns." Arg.string

